src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptions.vb

```diff
@@ -32,8 +44,110 @@ Namespace Microsoft.VisualStudio.Editors.OptionPages
             End Set
         End Property
 
+        ''' <summary>
+        ''' This setting has dynamic default value:
+        '''     - true if VS was installed from Preview channel
+        '''     - false if VS was installed from Release channel.
+        ''' When default setting value is changed by user, we write current value to a marker file 
+        ''' located under local AppData VS instance folder to be used by out-proc components.
+        ''' If there no marker file there, it measn VS is having default setting value which was 
+        ''' never changed by user manually. Out-proc components should treat absense of the file as 
+        ''' default option value and check the channel using same setup API we use here. 
+        ''' This absense=default logic is needed for the case VS is installed, was never started yet,
+        ''' but user runs msbuild form dev command line to build solution. In this case msbuild 
+        ''' components should have a way to figure out default setting on their own.
+        ''' 
+        ''' Note: This setting is not roamed. After chatting with shell team, we decided not to roam 
+        ''' it due to stability issues.
+        ''' </summary>
+        ''' <returns></returns>
+        Public Property UsePreviewSdk As Boolean
+            Get
+                Return If(_settingsManager?.GetValueOrDefault(UsePreviewSdkSettingKey, _isPreview), _isPreview)
+            End Get
+            Set
+                _settingsManager.SetValueAsync(UsePreviewSdkSettingKey, Value, isMachineLocal:=False)
+                UpdateSdkMarkerFile(Value)
+            End Set
+        End Property
+
         Public Sub New(serviceProvider As IServiceProvider)
             _settingsManager = CType(serviceProvider.GetService(GetType(SVsSettingsPersistenceManager)), ISettingsManager)
+            _shell = CType(serviceProvider.GetService(GetType(SVsShell)), IVsShell)
+
+            Dim oPath As Object = Nothing
+            If (Not VSErrorHandler.Failed(_shell.GetProperty(__VSSPROPID4.VSSPROPID_LocalAppDataDir, oPath))) And
+                (oPath IsNot Nothing) And
+                (TypeOf oPath Is String) Then
+                _localAppData = CType(oPath, String)
+            End If
+
+            _isPreview = IsPreview()
+        End Sub
+
+        ''' <summary>
+        ''' Uses VS setup API, to determine if current VS instance came from Preview channel or not.
+        ''' Note: we depend here on channel manifest format. If it changes, we could be broken. There is
+        ''' an integration test in WTE repo to be executed in nightly vendors' runs to protect form manifest 
+        ''' format changes.
+        ''' </summary>
+        ''' <returns></returns>
+        Private Function IsPreview() As Boolean
```

Comment:
Have we reached to setup for an API on this?

Reply:
+1. I'm not looking forward to duplicating this.

Reply:
Preview is just the name of the channel. Which could change , but if it is in the API name it would be fixed. 

What exactly are we duplicating here? We are analyzing a string property of the VS instance.

In any case, i can ping them , however do you really believe they would add this right now? I am not sure it would happen, let' see. 


---
In reply to: [191930452](https://github.com/dotnet/project-system/pull/3602#discussion_r191930452) [](ancestors = 191930452)

Reply:
> What exactly are we duplicating

This function. Resolver will need to do the same, right? I don't love copying all this. Nobody reading our source code will have any idea what WTE is or why that test is good enough reason to do this via string matching.

Reply:
You've written logic that is tied to the a technology we don't own or get notified when it changes, and is being tested by a set of tests that my team don't own or control while taking a dependency on an implementation detail. The first approach should be to talk to the team that owns this to ask for an API.